docs/docs/dev-guide-new-shield.md

```diff
@@ -98,6 +132,74 @@ this might look something like:
 };
 ```
 
+:::note
+For split keyboards, it is preferred to define only the `col-gpios` or `row-gpios` in the common shield .dtsi, depending on the diode-direction.
```

Comment:
```suggestion
For split keyboards, it is preferred to define only the `col-gpios` or `row-gpios` in the common shield .dtsi, depending on the `diode-direction` value.
```

Reply:
Also, before diving into the specifics of the matrix stuff, can we describe the `.dtsi` file and why we use it, how it is shared common DT settings that gets included into the specific left/right overlay files as a starting point?